cmd/metal-api/internal/service/v1/ip.go

```diff
@@ -41,6 +41,10 @@ type IPResponse struct {
 }
 
 func NewIPResponse(ip *metal.IP) *IPResponse {
+	tags := ip.Tags
```

Comment:
ip might be nil !

Reply:
Mh... yes, in theory this could happen. We do not check this in all these kind of mapping functions and just access the IP entity without any precautions. It would also be weird if it would be `nil`, but if you want I'll add a verification to all these functions?